go/tests/endpoint/endpoint.go

```diff
@@ -319,3 +318,82 @@ func Unzip(zipData []byte) error {
 
 	return nil
 }
+
+// NB: time.Duration is an int64 cast
+func Wait(dur time.Duration) {
+	if dur <= 0 { // default
+		Say("Waiting for 3 seconds")
+		time.Sleep(3 * time.Second)
+	} else {
+		Say(fmt.Sprintf("Waiting for %d seconds", dur))
+		time.Sleep(dur * time.Second)
+	}
+}
+
+func Write(filename string, contents []byte) error {
+	var err error
+	if socketPath != "" {
+		err = writeIPC(filename, contents)
```

Comment:
can we log which write we use here so we better understand the code path a test took?

Reply:
tried to address this in new commit with a Say() call indicating style of write to be performed just before the operation is performed